test/integration/evictions/evictions_test.go

```diff
@@ -0,0 +1,242 @@
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/apis/policy/v1beta1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_5"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/controller/disruption"
+	"k8s.io/kubernetes/pkg/controller/informers"
+	"k8s.io/kubernetes/pkg/util/intstr"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+const (
+	defaultTimeout = 10 * time.Minute
+)
+
+func rmSetup(t *testing.T) (*httptest.Server, *disruption.DisruptionController, cache.SharedIndexInformer, clientset.Interface) {
+	masterConfig := framework.NewIntegrationTestMasterConfig()
+	_, s := framework.RunAMaster(masterConfig)
+
+	config := restclient.Config{Host: s.URL}
+	clientSet, err := clientset.NewForConfig(&config)
+	if err != nil {
+		t.Fatalf("Error in create clientset: %v", err)
+	}
+	resyncPeriod := 12 * time.Hour
+	informers := informers.NewSharedInformerFactory(clientset.NewForConfigOrDie(restclient.AddUserAgent(&config, "pdb-informers")), nil, resyncPeriod)
+
+	rm := disruption.NewDisruptionController(
+		informers.Pods().Informer(),
+		clientset.NewForConfigOrDie(restclient.AddUserAgent(&config, "replicaset-controller")),
+	)
+	if err != nil {
+		t.Fatalf("Failed to create replicaset controller")
+	}
+	return s, rm, informers.Pods().Informer(), clientSet
+}
+
+func TestConcurrentEvictionRequests(t *testing.T) {
+	podNameFormat := "test-pod-%d"
+
+	s, rm, podInformer, clientSet := rmSetup(t)
+	defer s.Close()
+
+	ns := framework.CreateTestingNamespace("concurrent-eviction-requests", s, t)
+	defer framework.DeleteTestingNamespace(ns, s, t)
+
+	stopCh := make(chan struct{})
+	go podInformer.Run(stopCh)
+	go rm.Run(stopCh)
+
+	config := restclient.Config{Host: s.URL}
+	clientSet, err := clientset.NewForConfig(&config)
+
+	var gracePeriodSeconds int64 = 30
+	deleteOption := &v1.DeleteOptions{
+		GracePeriodSeconds: &gracePeriodSeconds,
+	}
+
+	// Create 10 pods to evict
+	for i := 0; i < 10; i++ {
+		pod := &v1.Pod{
+			ObjectMeta: v1.ObjectMeta{
+				Name:   fmt.Sprintf(podNameFormat, i),
+				Labels: map[string]string{"app": "test-evictions"},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:  "fake-name",
+						Image: "fakeimage",
+					},
+				},
+			},
+			Status: v1.PodStatus{
+				Phase: v1.PodRunning,
+				Conditions: []v1.PodCondition{
+					{
+						Type:   v1.PodReady,
+						Status: v1.ConditionTrue,
+					},
+				},
+			},
+		}
+
+		if _, err := clientSet.Core().Pods(ns.Name).Create(pod); err != nil {
+			t.Errorf("Failed to create pod: %v", err)
+		}
+
+		clientSet.Core().Pods(ns.Name).UpdateStatus(pod)
+	}
+
+	waitToObservePods(t, podInformer, 10)
+
+	pdb := &v1beta1.PodDisruptionBudget{
+		ObjectMeta: v1.ObjectMeta{
+			Name: fmt.Sprintf("test-pdb"),
+		},
+		Spec: v1beta1.PodDisruptionBudgetSpec{
+			MinAvailable: intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: 0,
```

Comment:
The intention of this test is to make sure PDB can handle at least a small number of concurrent eviction requests.
Using 0 here will generate more requests and make the rest of the test simple.

Test other aspects of evictions is beyond this test.